pkg/icingadb/db.go

```diff
@@ -228,13 +228,18 @@ func (db *DB) BuildWhere(subject interface{}) (string, int) {
 	return strings.Join(where, ` AND `), len(columns)
 }
 
+// OnSuccess is a callback for successful (bulk) DML operations.
+type OnSuccess[T any] func(ctx context.Context, affectedRows []T) (err error)
```

Comment:
Even for exactly one usage at the moment?

Reply:
It would still provide some documentation/context for what happens.

However, there are also places that wouldn't be covered by a single one of these. One could build something like follows, but I'm not sure if this wouldn't be too overengineered:
```go
BulkExec(..., OnSuccessMulti(
  OnSuccessIncrement(statsCtr),
  OnSuccessIncrement(telemetryCtr),
  OnSuccessForwardTo(ch),
))
```